src/main/java/frc/trigon/robot/misc/objectdetectioncamera/SimulationObjectDetectionCameraIO.java

```diff
@@ -40,127 +30,224 @@ protected SimulationObjectDetectionCameraIO(String hostname) {
+        for (Translation2d currentObject : objectsOnField) {
+            final Rotation2d robotYawToObject = getRobotYawToObject(currentObject, robotPose);
+            if (!isWithinHorizontalFOV(robotYawToObject, robotPose) || !isWithinDistance(objectsOnField.get(0), robotPose))
+                continue;
+            visibleObjects.add(currentIndex, currentObject);
+            currentIndex++;
         }
-        return posesArray;
+        return visibleObjects;
+    }
+
+    /**
+     * Gets the yaw of an object relative to the robot.
+     *
+     * @param objectPlacement the placement of the object on the field
+     * @param robotPose       the position of the robot on the field
+     * @return the yaw of the object relative to the robot
+     */
+    private Rotation2d getRobotYawToObject(Translation2d objectPlacement, Pose2d robotPose) {
+        final Translation2d robotToObject = objectPlacement.minus(robotPose.getTranslation());
+        return robotToObject.getAngle();
+    }
+
+    /**
+     * Checks if an object is within the field-of-view of the robot.
+     *
+     * @param objectYaw the yaw of the object relative to the camera
+     * @param robotPose the position of the robot on the field
+     * @return if the object is within the field-of-view of the robot
+     */
+    private boolean isWithinHorizontalFOV(Rotation2d objectYaw, Pose2d robotPose) {
+        return Math.abs(objectYaw.minus(robotPose.getRotation()).getRadians()) <= CAMERA_HORIZONTAL_FOV.getRadians() / 2;
     }
 
-    private boolean isEjecting() {
-        //todo: check if the robot is ejecting
-        return false;
+    /**
+     * Checks if an object is within the distance range of the robot.
+     *
+     * @param objectPlacement the placement of the object on the field
+     * @param robotPose       the position of the robot on the field
+     * @return if the object is withing the distance range of the robot
+     */
+    private boolean isWithinDistance(Translation2d objectPlacement, Pose2d robotPose) {
+        final double robotToObjectDistanceMeters = getObjectDistance(objectPlacement, robotPose);
+        return robotToObjectDistanceMeters <= MAXIMUM_VISIBLE_DISTANCE_METERS && robotToObjectDistanceMeters >= MINIMUM_VISIBLE_DISTANCE_METERS;
     }
 
-    private boolean isCollecting() {
-        //todo: check if the robot is collecting
-        return false;
+    /**
+     * Gets the distance between the robot and an object.
+     *
+     * @param objectPlacement the placement of the object on the field
+     * @param robotPose       the position of the robot on the field
+     * @return the distance between the robot and the object
+     */
+    private double getObjectDistance(Translation2d objectPlacement, Pose2d robotPose) {
+        final Translation2d robotToObject = objectPlacement.minus(robotPose.getTranslation());
+        return robotToObject.getNorm();
```

Comment:
we can use the getDistance function of translation2d. maybe change the name of the function a bit
this calculates, not get